src/Modules/UngradedModules/FreeResolutions.jl

```diff
@@ -173,6 +173,58 @@ function _extend_free_resolution(cc::Hecke.ComplexOfMorphisms, idx::Int)
     return map(cc, first(r))
   end
 
+  # The code further below is incorrect, see #4998. Until we can provide a 
+  # reasonable fix using a continuation of Schreyer's ordering, we 
+  # provide a patch with new initiation of kernel computations.
+
+  phi = first(cc.maps)
+  K, inc_K = kernel(phi)
+  R = base_ring(K)
+  F = ambient_free_module(K)
+  SF = singular_module(F)
+  SK = singular_generators(algorithm == :fres ? groebner_basis(K) : K.sub.gens) # We need to start with a gb for `fres`.
+  
+  if is_one(len_missing)
+    cod = domain(first(cc.maps))
+    phi = is_graded(cod) ? graded_map(cod, ambient_representatives_generators(K); check=false) : hom(free_module(R, ngens(K)), cod, ambient_representatives_generators(K))
+    pushfirst!(cc.maps, phi)
+    if is_zero(K)
+      cc.complete=true
+    end
+    return first(cc.maps)
+  end
+
+  if algorithm == :fres
+    res = Singular.fres(SK, len_missing, "complete")
+  elseif algorithm == :lres
+    error("LaScala's method is not yet available in Oscar.")
+  elseif algorithm == :mres
+    res = Singular.mres(SK, len_missing)
+  elseif algorithm == :nres
+    res = Singular.nres(SK, len_missing)
+  else
+    error("Unsupported algorithm $algorithm")
+  end
+
+  for j in 1:Singular.length(res)
+    cod = domain(first(cc.maps))
+    I = SubModuleOfFreeModule(cod, res[j]) # convert to an Oscar module
+    phi = is_graded(cod) ? graded_map(cod, gens(I); check=false) : hom(free_module(R, ngens(I)), cod, gens(I))
+    pushfirst!(cc.maps, phi)
+    if is_zero(I)
+      cc.complete = true
+      return first(cc.maps)
+    end
+  end
+  if is_zero(res[Singular.length(res)+1])
+    cod = domain(first(cc.maps))
+    pushfirst!(cc.maps, is_graded(cod) ? graded_map(cod, elem_type(cod)[]; check=false) : hom(free_module(R, 0), cod, elem_type(cod)[]))
+    cc.complete = true
+    return cc.maps[2]
+  end
+  return first(cc.maps)
+
+  #=
```

Comment:
please remove the code block below instead of commenting it out. That makes it way easiert to understand the diff in the github diff view